src/main/ruby/truffleruby/core/type.rb

```diff
@@ -145,8 +145,12 @@ def self.rb_num2dbl(val)
         val.to_f
       elsif Primitive.is_a?(val, Rational)
         val.to_f
+      elsif val === true
```

Comment:
`===` seems wrong here. I guess you meant to check by identity or more strictly (i.e., JS's `===`)?
Either way this should use `Primitive.true?(val)` (and same for `false` below)